code/modules/mob/living/navigation.dm

```diff
@@ -0,0 +1,70 @@
+#define MAX_NAVIGATE_RANGE 125
+
+/mob/living
+	COOLDOWN_DECLARE(navigate_cooldown)
+
+/mob/living/verb/navigate()
```

Comment:
you cant do all of this within a verb because verbs are executed as the last step of the tick if sent via client commands which means that they cause a lot of overtime. you need to defer this work to the next tick at least. since biddle verbs isnt complete yet you can just use a timer with duration world.tick_lag and offload the majority of this verb to the proc called by that timer